### PR TITLE
feature: v1 proposed layout

### LINKED ISSUE #58: feat: v1 UI
# V1

pqp front-end é chato pra caralho

This issue tracks the implementation of V1, focusing on UI enhancements and feature completions. Below is a detailed todo list of the UI components and their current status.

## Screens

- [x]  Theme Setup

- Authenticate
  - [x] UI
  - [x] Light mode
  - [x] Dark mode
  - [ ] I18n
- Settings
  - [x] UI
  - [x] Light mode
  - [x] Dark mode
  - [ ] I18n
- Stats
  - [x] UI
  - [x] Light mode
  - [x] Dark mode
  - [ ] I18n
- About us
  - [x] UI
  - [x] Light mode
  - [x] Dark mode
  - [ ] I18n
- Sidebar
  - [x] UI
  - [x] Light mode
  - [x] Dark mode
  - [ ] I18n

### PR DESCRIPTION
**Pull request by moovimentos**
# Motivation

pqp front-end é chato pra caralho

This Pull Request tracks the implementation of V1, focusing on UI enhancements and feature completions. Below is a detailed todo list of the UI components and their current status.

## Screens

- [x]  Theme Setup

- Authenticate
  - [x] UI
  - [x] Light mode
  - [x] Dark mode
  - [ ] I18n
- Settings
  - [x] UI
  - [x] Light mode
  - [x] Dark mode
  - [ ] I18n
- Stats
  - [x] UI
  - [x] Light mode
  - [x] Dark mode
  - [ ] I18n
- About us
  - [x] UI
  - [x] Light mode
  - [x] Dark mode
  - [ ] I18n
- Sidebar
  - [x] UI
  - [x] Light mode
  - [x] Dark mode
  - [ ] I18n
  
Closes https://github.com/basementdevs/twitch-better-profile/issues/58

### CODE REVIEW

File: src/resources/components/app/main-content.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import Theme from "../theme/theme";
+
+type MainContentProps = {
+  selectedItem: string;
+  userService: UserStorageService;
+};
+
+const MainContent = ({ selectedItem, userService }: MainContentProps) => {
+  switch (selectedItem) {
+    case "Configurações":
```
Comment (danielhe4rt) [inline src/resources/components/app/main-content.tsx]: Cases with non-ascii characters isn't the best of the ideas. Any reason for it being this way?

File: src/resources/components/app/profile-ribbon.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import type UserStorageService from "~services/user/user-storage-service";
+import type { User } from "~types/types";
+
+type ProfileRibbonType = {
+  user: User;
+};
+
+function ProfileRibbon({ user }: ProfileRibbonType) {
+  const onStorageClear = async () => {
+    const { Storage } = await import("@plasmohq/storage");
```
Comment (danielhe4rt) [inline src/resources/components/app/profile-ribbon.tsx]: Stop using drugs and code, please.

File: src/resources/components/settings/settings-form.tsx [hunk 3]
```diff
@@ -1,53 +1,66 @@
...
 
+    const payload = {
+      pronouns: selectedPronoun,
+      locale: navigator.language,
+      occupation_id: selectedOccupation,
+      user_id: twitchUser.id,
+      username: twitchUser.login,
+    };
+
     const response = await fetch(
```
Comment (danielhe4rt) [inline src/resources/components/settings/settings-form.tsx:64]: Moving this to Services in the future.

File: src/services/user/user-consumer-service.ts [hunk 4]
```diff
@@ -53,7 +53,9 @@ export async function authenticateWithServer(code: string) {
 
   const data: { user: User; authorization: AccessTokenResponse } =
     await response.json();
+  console.log(data);
```
Comment (danielhe4rt) [inline src/services/user/user-consumer-service.ts]: Remove it.

**Review by danielhe4rt — APPROVED** [review]
You're amazing man! Please ask for a raise in your current company and tells that it was a request from me.

It will work, trust me!

File: src/resources/components/app/main-content.tsx (hunk 1 above)
Comment (moovimentos) [reply src/resources/components/app/main-content.tsx]: It was the quickest way to implement the feature, but I'm going to switch the keys to English.

**Review by danielhe4rt — APPROVED** [review]

**Pull request merged**